src/libAtomVM/port.h

```diff
@@ -83,6 +82,22 @@ static inline void port_send_reply(Context *ctx, term pid, term ref, term payloa
     port_send_message(ctx->global, pid, reply);
 }
 
+typedef struct
+{
+    term req;
+
+    term pid;
+    term ref;
+} GenMessage;
```

Comment:
Not all messages will have a ref or even a pid.  Are we okay with this?  I am fine with it, and I consider this a kind of interim solution, so that's fine.  Just thinking ahead to what an `$info` or `$cast` message would look like.

This is just an observation, not a change request.

Reply:
In that case `GenCastMessage` instead of `GenCallMessage` will be returned (pid and ref will be invalid).